frontend/public/components/modals/promote-crd-modal.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { FLAGS } from '../../features';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import { k8sCreate, referenceForCRD } from '../../module/k8s/';
+import { ConsoleExtensionModel } from '../../models';
+import { PromiseComponent, Dropdown } from '../utils';
+const constructConsoleExtensionInstance = (crd, navSection) => {
+  const displayName = _.get(crd, ['metadata', 'annotations', 'displayName'], crd.spec.names.kind);
+  return {
+    apiVersion: 'config.openshift.io/v1',
+    kind: 'ConsoleExtension',
+    metadata: {
+      name: `${crd.spec.names.singular}-console-extension`
+    },
+    spec: {
+      navExtension: {
+        displayName: displayName,
+        navSection: navSection,
+      },
+      scope: crd.spec.scope,
+      reference: referenceForCRD(crd),
+    }
+  };
+};
+export class PromoteCrdModal extends PromiseComponent {
+  constructor(props) {
+    super(props);
+    this.state = Object.assign(this.state, {
+      section: 'Home',
+      displayName: this.props.crd.spec.names.kind,
+    });
+    this._submit = this._submit.bind(this);
+    this._onSectionChange = this._onSectionChange.bind(this);
+    this._onDisplayNameChange = this._onDisplayNameChange.bind(this);
+  }
+  _submit(event) {
+    event.preventDefault();
+    const consoleExtension = constructConsoleExtensionInstance(this.props.crd, this.state.section);
+    this.handlePromise(k8sCreate(ConsoleExtensionModel, consoleExtension)).then(() => this.props.close());
+  }
+  _onSectionChange(section) {
+    this.setState({
+      section: section
+    });
+  }
+  _onDisplayNameChange(event) {
+    this.setState({
+      displayName: event.target.value
+    });
+  }
+  render() {
+    const sections = ['Home', 'Operators', 'Workloads', 'Networking', 'Storage', 'Builds', 'Service Catalog', 'Monitoring', 'Administration'];
+    const sectionsObject = _.reduce(sections, (acc, section) => {
+      if (section === 'Operators' && !FLAGS.OPERATOR_LIFECYCLE_MANAGER) {
+        return;
+      }
+      if (section === 'Builds' && !FLAGS.OPENSHIFT) {
```

Comment:
This doesn't capture all the checks. I'm a little hesitant to have the logic for what nav sections are shown when in multiple places. I think I'd rather show all options.